core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -398,6 +399,34 @@
                                  system-from
                                  nil))))))))))
 
+(def ^:private neg-or-zero-int? (complement pos-int?))
+
+(defn- iid-selector [^ByteBuffer iid-bb]
+  (reify IRelationSelector
+    (select [_ _allocator rel-rdr _params]
+      (let [ptr (ArrowBufPointer.)
+            iid-rdr (.readerForName rel-rdr "xt$iid")
+            value-count (.valueCount iid-rdr)]
+        ;; lower-bound
+        (loop [left 0 right (dec value-count)]
```

Comment:
think this might be broken for an empty rel? worth a unit test.

i.e. left 0, right -1 then doesn't hit your `(= left right)` case to fall out of the loop.

also nit: don't mind this being on one line, but prefer a comma to visually separate the two bindings: `[left 0, right (dec value-count)]`